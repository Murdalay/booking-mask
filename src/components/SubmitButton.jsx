import styled from 'styled-components';
import { Button, utils } from 'styled-minimal';

const { spacer } = utils;

const SubmitButton = styled(Button)`
  font-size: 1.6rem;
  padding: ${spacer(2)};
  border-radius: 0;
  display: table-cell;
  min-height: ${props => props.theme.sizes.controlHeight}px;
  width: 100%;

  &:hover {
    opacity: 0.9;
    width: inherit;
    transform: none;
  }

  &.active {
    color: #000;
  }

  span {
    display: inline-block;
    margin-right: 0.4rem;
    text-transform: uppercase;
    position: relative;
  }
`;

export default SubmitButton;
