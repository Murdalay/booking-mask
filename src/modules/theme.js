export const rowHeight = 70;
export const controlHeight = 55;
export const appColor = '#00b4d5';
export const secondaryColor = '#88e4f5';
export const actionColor = '#66b4e5';
export const inputMinWidth = 150;
export const defaultIconColor = '#444';
export const defaultTextColor = '#555';
export const borderColor = 'rgb(204, 204, 204)';
export const easing = 'cubic-bezier(0.35, 0.01, 0.77, 0.34);';
export const inputFontSize = '1.5rem';
export const inputBackgroundColor = '#fff';

export default {
  breakpoints: [400, 768, 1024, 1280, 1920],
  palette: {
    primary: appColor,
    secondary: secondaryColor,
    actionColor,
    appColor,
    borderColor,
    defaultTextColor,
    defaultIconColor,
    inputBackgroundColor,
  },
  sizes: {
    rowHeight,
    controlHeight,
    borderColor,
    inputMinWidth,
    inputFontSize,
  },
  input: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
      default: 0,
    },
  },
  button: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
    },
    padding: {
      lg: [12, 28],
      xl: [14, 32],
    },
  },
};
