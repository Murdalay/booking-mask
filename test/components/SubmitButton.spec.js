import React from 'react';
import theme from 'modules/theme';

import SubmitButton from 'components/SubmitButton';

describe('SubmitButton', () => {
  const wrapper = mount(<SubmitButton theme={theme} />);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
