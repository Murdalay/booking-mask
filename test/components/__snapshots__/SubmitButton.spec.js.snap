// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #00b4d5;
  border: 1px solid #00b4d5;
  color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  padding: 9px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
  font-size: 1.6rem;
  padding: 8px;
  border-radius: 0;
  display: table-cell;
  min-height: 55px;
  width: 100%;
}

.c0:disabled {
  pointer-events: none;
}

.c0:hover {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c0:focus {
  outline-color: #00b4d5;
}

.c0 .c1 {
  margin-left: 5px;
}

.c0:hover {
  opacity: 0.9;
  width: inherit;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c0.active {
  color: #000;
}

.c0 span {
  display: inline-block;
  margin-right: 0.4rem;
  text-transform: uppercase;
  position: relative;
}

<SubmitButton
  animate={false}
  as="button"
  block={false}
  disabled={false}
  outline={false}
  size="md"
  theme={
    Object {
      "breakpoints": Array [
        400,
        768,
        1024,
        1280,
        1920,
      ],
      "button": Object {
        "borderRadius": Object {
          "lg": 0,
          "xl": 0,
          "xs": 0,
        },
        "padding": Object {
          "lg": Array [
            12,
            28,
          ],
          "xl": Array [
            14,
            32,
          ],
        },
      },
      "input": Object {
        "borderRadius": Object {
          "default": 0,
          "lg": 0,
          "xl": 0,
          "xs": 0,
        },
      },
      "palette": Object {
        "actionColor": "#66b4e5",
        "appColor": "#00b4d5",
        "borderColor": "rgb(204, 204, 204)",
        "defaultIconColor": "#444",
        "defaultTextColor": "#555",
        "inputBackgroundColor": "#fff",
        "primary": "#00b4d5",
        "secondary": "#88e4f5",
      },
      "sizes": Object {
        "borderColor": "rgb(204, 204, 204)",
        "controlHeight": 55,
        "inputFontSize": "1.5rem",
        "inputMinWidth": 150,
        "rowHeight": 70,
      },
    }
  }
  type="button"
  variant="primary"
>
  <button
    className="c0"
    disabled={false}
    size="md"
    type="button"
  />
</SubmitButton>
`;
